src/com/intellij/idea/plugin/hybris/runConfigurations/HybrisJUnitExtension.kt

```diff
@@ -50,22 +54,61 @@ class HybrisJUnitExtension : RunConfigurationExtension() {
 
         addVmParameterIfNotExist(vmParameters, "-ea")
 
+        addPlatformHome(vmParameters, project)
+        addJavaRunProperties(project, vmParameters)
+        addHybrisDirProperties(project, vmParameters)
+
+        enhanceClassPath(params, project)
+    }
+
+    private fun enhanceClassPath(params: JavaParameters, project: Project) {
+        val classPathEntries = HashSet<String>()
+
+        val modules: Array<Module> = ModuleManager.getInstance(project).modules
+        for (module in modules) {
+            // Get the module's output paths (both production and test)
+            val moduleRootManager = ModuleRootManager.getInstance(module)
+
+            // Get the compiler output paths for production and test
+            val productionOutput = moduleRootManager.getModuleExtension(CompilerModuleExtension::class.java)?.compilerOutputPath
+            val testOutput = moduleRootManager.getModuleExtension(CompilerModuleExtension::class.java)?.compilerOutputPathForTests
+
+            // Add the output paths to the classpath
+            if (productionOutput != null && classPathEntries.add(productionOutput.path)) {
+                params.classPath.add(productionOutput.path)
+            }
+            if (testOutput != null && classPathEntries.add(testOutput.path)) {
+                params.classPath.add(testOutput.path)
+            }
+
+            // **Add module dependencies to classpath**
+            OrderEnumerator.orderEntries(module).recursively().classes().roots.forEach {
+                val path = it.presentableUrl
+                if (classPathEntries.add(path)) {
+                    params.classPath.add(it)
+                }
+            }
+        }
+    }
+
+    private fun addPlatformHome(vmParameters: ParametersList, project: Project) {
         if (vmParameters.parameters.none { it.startsWith("-Dplatformhome=") }) {
             HybrisRootUtil.findPlatformRootDirectory(project)
                 ?.path
                 ?.let { vmParameters.add("-Dplatformhome=$it") }
         }
+    }
 
-        if (!params.env.containsKey(ENV_HYBRIS_DATA_DIR)) {
-            val settings = ProjectSettingsComponent.getInstance(project).state
-            val hybrisDataDirPath = FileUtil.toCanonicalPath(
-                "${project.basePath}/${settings.hybrisDirectory}/$HYBRIS_DATA_DIRECTORY"
-            )
-
-            if (File(hybrisDataDirPath).exists()) {
-                params.addEnv(ENV_HYBRIS_DATA_DIR, hybrisDataDirPath)
+    private fun addJavaRunProperties(project: Project, vmParameters: ParametersList) {
+        PropertyService.getInstance(project)
+            ?.findProperty(PROPERTY_STANDALONE_JAVAOPTIONS)
+            ?.let { property -> StringTokenizer(property.trim { it <= ' ' }) }
+            ?.let {
+                while (it.hasMoreTokens()) {
+                    val newParam = PATTERN.matcher(it.nextToken()).replaceAll("")
```

Comment:
it would be better to have `PATTERN` renamed to something more self-descriptive.